src/screenComponents/missileTubeControls.cpp

```diff
@@ -90,14 +93,21 @@ void GuiMissileTubeControls::onUpdate()
         case MissileTubes::MountPoint::State::Empty:
             rows[n].load_button->setEnable(tube.canLoad(load_type));
             rows[n].load_button->setText(tr("missile","Load"));
+            if (health <= 0)
+                rows[n].load_button->disable();
             rows[n].fire_button->disable()->show();
             rows[n].fire_button->setText(getTubeName(tube.direction) + ": " + tr("missile","Empty"));
             rows[n].loading_bar->hide();
             break;
         case MissileTubes::MountPoint::State::Loaded:
-            rows[n].load_button->enable();
             rows[n].load_button->setText(tr("missile","Unload"));
-            rows[n].fire_button->enable()->show();
+            if ((health <= 0) || (power_level <=0)) {
+                rows[n].fire_button->disable();
+                rows[n].load_button->disable();
```

Comment:
the lacking of show might cause a problem in some edge cases, like opening the weapon screen when the system is broken.

Reply:
Hope that makes things a bit clearer?

Reply:
Honestly, the code is a bit messy. And there might be edge cases where this currently would go wrong still :)